File: Project_24/LifeTracker/backend/src/main/java/com/lifetracker/modules/expenses/dto/CreateExpenseRequest.java
package com.lifetracker.modules.expenses.dto;

import com.lifetracker.modules.expenses.enums.ExpenseType;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.math.BigDecimal;
import java.time.LocalDate;

public record CreateExpenseRequest(
        @NotNull ExpenseType expenseType,
        @NotBlank String category,
        @NotBlank String title,
        String description,
        @NotNull @DecimalMin("0.01") BigDecimal amount,
        @NotNull LocalDate expenseDate,
        String paymentMode,
        String notes
) {
}

File: Project_24/LifeTracker/frontend/lib/models/create_expense_request.dart
import 'expense_type.dart';

class CreateExpenseRequest {
  const CreateExpenseRequest({
    required this.expenseType,
    required this.category,
    required this.title,
    this.description,
    required this.amount,
    required this.expenseDate,
    this.paymentMode,
    this.notes,
  });

  final ExpenseType expenseType;
  final String category;
  final String title;
  final String? description;
  final double amount;
  final DateTime expenseDate;
  final String? paymentMode;
  final String? notes;

  Map<String, dynamic> toJson() {
    return {
      'expenseType': expenseType.apiValue,
      'category': category,
      'title': title,
      'description': description,
      'amount': amount,
      'expenseDate': _formatDate(expenseDate),
      'paymentMode': paymentMode,
      'notes': notes,
    };
  }

  String _formatDate(DateTime date) => date.toIso8601String().split('T').first;
}
